Bug 1493597 - Funnelcake 138 & 139

nthomas-mozilla opened the pull request:
Mike, these configs should do the trick for 138 (control) and 139 (test). Romain and I discussed (on the gdoc) having a full installer for the control so we're comparing applies with apples retention-wise. I'll need to add the url to the custom stub once the patch has reviewed and I've built it on mozilla-release.

In other news, the parameters which no longer do anything have been cleaned out - `aus, dist_id, dist_version, s3cfg, bucket,` and `output_dir`. RE the last, we no longer need to bump the config for a respin. Would a mozilla-partners/config_template repo, storing the latest minimal config, be useful for quickly creating new configs ?

mkaply commented on desktop/funnelcake138/distribution/distribution.ini:
```diff
@@ -0,0 +1,10 @@
+
+[Preferences]
+app.partner.mozilla138="mozilla138"
+startup.homepage_welcome_url="https://www.mozilla.org/%LOCALE%/%APP%/%VERSION%/firstrun/?f=138"
```
This can be removed. We've switched to about:welcome internally, so we no longer do welcome URLs in the funnelcake.

nthomas-mozilla replied:
Will do both of these.

mkaply commented on desktop/funnelcake138/repack.cfg:
```diff
@@ -0,0 +1,10 @@
+linux-i686=false
+linux-x86_64=false
+mac=false
+win32=false
```
Are we sure we will never have a 32 bit download?

nthomas-mozilla replied:
The argument is there are much fewer win32 installs per day so it was more complexity for little gain.

mkaply commented on desktop/funnelcake139/distribution/distribution.ini:
```diff
@@ -0,0 +1,10 @@
+
+[Preferences]
+app.partner.mozilla139="mozilla139"
+startup.homepage_welcome_url="https://www.mozilla.org/%LOCALE%/%APP%/%VERSION%/firstrun/?f=139"
```
Same comment as before, remove this line.

mkaply commented on desktop/funnelcake139/repack.cfg:
```diff
@@ -0,0 +1,13 @@
+win64=true
+
+# custom installer
+replacement_setup_exe="TBD"
```
Do we need a reminder to change this?

nthomas-mozilla replied:
Now included in the latest revision.

mkaply commented:
> In other news, the parameters which no longer do anything have been cleaned out - `aus, dist_id, dist_version, s3cfg, bucket,` and `output_dir`. RE the last, we no longer need to bump the config for a respin. Would a mozilla-partners/config_template repo, storing the latest minimal config, be useful for quickly creating new configs ?

so it output_dir generated automatically using the correct path? Was it always that way, we just added a custom path because of the v1/v2 thing?

I love the idea of a config template repo.

nthomas-mozilla commented:
> so it output_dir generated automatically using the correct path? Was it always that way, we just added a custom path because of the v1/v2 thing?

Since we moved everything into taskcluster it's controlled by code in gecko and a component called beetmover. There's a parameter called `release_partner_build_number` which turns into the vN in the publicly hosted builds. If defaults to 1 for the automation, and we make sure to increase that each time there's a respin.

> I love the idea of a config template repo.

Cool, could you set one up ?

nthomas-mozilla commented:
Is the patch looking OK now ?
